In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


# Expression_df 데이터프레임

In [2]:
# 압축 해제
import gzip
import shutil

with gzip.open("/content/drive/MyDrive/BIO_MEDICAL_PROJECT/obesity_gene_analysis/data/GSE77962_series_matrix.txt.gz", 'rb') as f_in:
    with open("/content/drive/MyDrive/BIO_MEDICAL_PROJECT/obesity_gene_analysis/data/GSE77962_series_matrix.txt", 'wb') as f_out:
        shutil.copyfileobj(f_in, f_out)

In [6]:
import pandas as pd
GSE77962_series_matrix_df = pd.read_csv("/content/drive/MyDrive/BIO_MEDICAL_PROJECT/obesity_gene_analysis/data/GSE77962_series_matrix.txt", sep="\t", comment='!')

In [19]:
GSE77962_series_matrix_df

ID_REF  GSM2062466  GSM2062467  GSM2062468  GSM2062469  GSM2062470  \
0      7892501    4.428059    4.175376    4.025431    4.058749    5.077221   
1      7892502    3.015303    3.382805    4.266405    3.859811    3.524342   
2      7892503    2.469030    1.683800    1.353440    1.833336    1.773266   
3      7892504    7.469548    7.575583    8.319473    7.877318    7.463798   
4      7892505    2.655132    1.349682    1.610853    1.264390    2.165488   
...        ...         ...         ...         ...         ...         ...   
33292  8180414    3.771268    3.827958    4.060053    4.299973    4.432689   
33293  8180415    7.106688    6.772125    6.657147    7.350512    7.153225   
33294  8180416    3.159829    3.384084    3.988990    3.447247    3.426313   
33295  8180417    7.572312    7.260562    7.102751    7.543385    7.276615   
33296  8180418    8.086955    7.506301    7.384701    7.924522    7.381025   

       GSM2062471  GSM2062472  GSM2062473  GSM2062474  ...  GSM2062608  \
0        4.167342    5.217643    5.940471    4.454951  ...    4.535857   
1        3.983906    3.177080    3.634795    2.864439  ...    4.340592   
2        2.073515    2.035071    1.663281    1.647068  ...    2.233719   
3        7.777693    8.129641    7.505768    8.378240  ...    8.569401   
4        1.744837    2.018746    1.284328    1.381185  ...    1.300862   
...           ...         ...         ...         ...  ...         ...   
33292    4.450297    4.603566    4.273023    4.684982  ...    4.710916   
33293    7.043244    7.231078    6.428579    6.299577  ...    6.784532   
33294    3.424887    3.423193    3.636731    3.205910  ...    3.607290   
33295    7.051876    7.267355    6.877710    6.694160  ...    7.217937   
33296    7.339969    7.359909    7.207605    6.963106  ...    7.129784   

       GSM2062609  GSM2062610  GSM2062611  GSM2062612  GSM2062613  GSM2062614  \
0        4.908356    5.564917    2.662255    5.434389    4.318741    5.142372   
1        3.783674    4.117283    3.957558    3.836091    3.138714    4.347246   
2        2.339964    2.191594    2.925588    1.825886    2.154871    2.842330   
3        8.639623    8.087239    8.292656    7.683577    7.317489    7.908107   
4        1.628132    1.477364    1.657482    1.028516    1.138573    1.237365   
...           ...         ...         ...         ...         ...         ...   
33292    4.237877    4.360936    4.887058    4.566632    4.723105    4.759401   
33293    6.962716    6.634560    6.862306    6.961862    6.817866    6.760013   
33294    3.704066    3.750592    3.502732    3.150830    3.621640    3.182532   
33295    6.992108    7.343185    6.941972    7.337702    7.330790    7.255967   
33296    6.373951    6.973372    6.545439    7.355312    6.848314    7.308610   

       GSM2062615  GSM2062616  GSM2062617  
0        5.739793    5.183112    5.107738  
1        4.117830    4.315688    3.758828  
2        2.123143    2.222103    1.485676  
3        7.973787    7.984416    7.821190  
4        1.726360    1.865871    1.279610  
...           ...         ...         ...  
33292    5.170525    4.519322    3.996389  
33293    6.856684    7.077445    7.054198  
33294    3.449176    3.040495    3.080801  
33295    7.227201    7.187702    7.173843  
33296    6.973893    7.526776    6.731832  

[33297 rows x 153 columns]

| 항목             | 의미                                                          |
| -------------- | ----------------------------------------------------------- |
| **행 (33297개)** | 각각 하나의 **유전자 프로브** (ID\_REF). 즉, 유전자 표현형에 대응하는 탐침(probe) ID |
| **열 (153개)**   | 각각 하나의 **샘플 (GSM ID)** = 피실험자들의 **조직 샘플 (예: 지방조직)**         |


In [20]:
expression_df=GSE77962_series_matrix_df

In [21]:
expression_df.to_csv("/content/drive/MyDrive/BIO_MEDICAL_PROJECT/obesity_gene_analysis/data/expression_df.csv")

# Platform_df 데이터프레임

In [9]:
# txt파일 처음 읽어보고 데이터 파악하기
platform_path = "/content/drive/MyDrive/BIO_MEDICAL_PROJECT/obesity_gene_analysis/data/GPL11532-32230.txt"

with open(platform_path, 'r') as f:
    for i in range(40):
        line = f.readline()
        print(f"{i}: {line.strip()}")

0: #ID = transcript_cluster_id
1: #GB_LIST = GenBank and RefSeq Accessions from mrna_assignment column
2: #SPOT_ID = genomic location of the transcript cluster in the version of the genome assembly used at annotation time. Coordinates are standard 1-based (length=stop-start+1).
3: #seqname = chromosome number
4: #RANGE_GB = NCBI RefSeq for chromosome of current build
5: #RANGE_STRAND = strand  (+|-)
6: #RANGE_START = start  (integer). Coordinates are standard 1-based (length=stop-start+1).
7: #RANGE_STOP = stop  (integer). Coordinates are standard 1-based (length=stop-start+1).
8: #total_probes = Total number of probes contained by this transcript cluster.
9: #gene_assignment = Gene information for each assigned mRNA for mRNAs that corresponds to known genes (multipart).
10: #mrna_assignment = Description of the public mRNAs that should be detected by the sets within this transcript cluster based on sequence alignment (multipart).
11: #category = Array design category of the transcript

In [15]:
# 0~12행까지 주석이었으니 skiprows=12을 통해 정확한 헤더 위치를 지정
GPL11532_32230_df = pd.read_csv(platform_path, sep="\t", skiprows=12, low_memory=False)

In [16]:
GPL11532_32230_df

ID                                            GB_LIST  \
0      7896736                                                NaN   
1      7896738                                                NaN   
2      7896740  NM_001005240,NM_001004195,NM_001005484,BC13684...   
3      7896742                                  BC118988,AL137655   
4      7896744  NM_001005277,NM_001005221,NM_001005224,NM_0010...   
...        ...                                                ...   
33292  8180416                                        NM_002551.3   
33293  8180417                                        NM_018396.2   
33294  8180418                                     NM_001040145.1   
33295  7976552                                                NaN   
33296  8020352                                                NaN   

                       SPOT_ID seqname      RANGE_GB RANGE_STRAND  \
0             chr1:53049-54936    chr1  NC_000001.10            +   
1             chr1:63015-63887    chr1  NC_000001.10            +   
2             chr1:69091-70008    chr1  NC_000001.10            +   
3           chr1:334129-334296    chr1  NC_000001.10            +   
4           chr1:367659-368597    chr1  NC_000001.10            +   
...                        ...     ...           ...          ...   
33292         flmrna->unmapped     ---           NaN          ---   
33293         flmrna->unmapped     ---           NaN          ---   
33294         flmrna->unmapped     ---           NaN          ---   
33295  chr14:96154516-96154596   chr14   NC_000014.8            +   
33296  chr18:14341593-14342513   chr18   NC_000018.9            +   

       RANGE_START  RANGE_STOP  total_probes  \
0            53049       54936             7   
1            63015       63887            31   
2            69091       70008            24   
3           334129      334296             6   
4           367659      368597            36   
...            ...         ...           ...   
33292            0           0            39   
33293            0           0            33   
33294            0           0             6   
33295     96154516    96154596             4   
33296     14341593    14342513             9   

                                         gene_assignment  \
0                                                    ---   
1                                                    ---   
2      NM_001005240 // OR4F17 // olfactory receptor, ...   
3      ENST00000388975 // SEPT14 // septin 14 // 7p11...   
4      NM_001005277 // OR4F16 // olfactory receptor, ...   
...                                                  ...   
33292                                                ---   
33293                                                ---   
33294                                                ---   
33295                                                ---   
33296                                                ---   

                                         mrna_assignment          category  
0                                                    ---               ---  
1      ENST00000328113 // ENSEMBL // cdna:pseudogene ...              main  
2      NM_001005240 // RefSeq // Homo sapiens olfacto...              main  
3      ENST00000388975 // ENSEMBL // Septin-14 gene:E...              main  
4      NM_001005277 // RefSeq // Homo sapiens olfacto...              main  
...                                                  ...               ...  
33292  NM_002551.3 // --- // Homo sapiens olfactory r...  flmrna->unmapped  
33293  NM_018396.2 // --- // Homo sapiens methyltrans...  flmrna->unmapped  
33294  NM_001040145.1 // --- // Homo sapiens chromoso...  flmrna->unmapped  
33295                                                ---               ---  
33296                                                ---               ---  

[33297 rows x 12 columns]

| 컬럼명                  | 의미 설명                                                               |
| -------------------- | ------------------------------------------------------------------- |
| **ID**               | probe ID (예: `7896738`, `7896817`) → `series matrix`의 `ID_REF`와 매칭됨 |
| **GB\_LIST**         | GenBank accession number (예: `NM_001005240`, `BC118988` 등)          |
| **SPOT\_ID**         | 위치 정보 (예: `chr1:901877-910488`)                                     |
| **seqname**          | 해당 유전자가 위치한 염색체명 (예: `chr1`, `chr14`)                               |
| **RANGE\_GB**        | GenBank 염색체 위치 정보 (예: `NC_000001.10`)                               |
| **RANGE\_STRAND**    | DNA 가닥 방향 (`+` 또는 `-`)                                              |
| **RANGE\_START**     | 유전자 시작 위치 (예: `53049`)                                              |
| **RANGE\_STOP**      | 유전자 종료 위치 (예: `54936`)                                              |
| **total\_probes**    | 해당 probe에 할당된 총 탐침 수                                                |
| **gene\_assignment** | 유전자 주석 정보. `//` 사이에 유전자 이름이 들어 있으며, 복수 가능                           |
| **mrna\_assignment** | mRNA 정보 (`ENSEMBL`, `RefSeq`, `GenBank` 등)                          |
| **category**         | 주석 타입 (예: `main`, `flmrna->unmapped` 등)                             |


In [17]:
import re

# 예시 DataFrame: GPL11532_32230_df
# gene_assignment에서 유전자 이름 추출
def extract_gene_symbol(text):
    if pd.isna(text):
        return None
    match = re.search(r'//\s*(.*?)\s*//', text)
    return match.group(1) if match else None

# 새 열 추가
platform_df= GPL11532_32230_df.copy()
platform_df['gene_symbol'] = GPL11532_32230_df['gene_assignment'].apply(extract_gene_symbol)

In [18]:
platform_df

ID                                            GB_LIST  \
0      7896736                                                NaN   
1      7896738                                                NaN   
2      7896740  NM_001005240,NM_001004195,NM_001005484,BC13684...   
3      7896742                                  BC118988,AL137655   
4      7896744  NM_001005277,NM_001005221,NM_001005224,NM_0010...   
...        ...                                                ...   
33292  8180416                                        NM_002551.3   
33293  8180417                                        NM_018396.2   
33294  8180418                                     NM_001040145.1   
33295  7976552                                                NaN   
33296  8020352                                                NaN   

                       SPOT_ID seqname      RANGE_GB RANGE_STRAND  \
0             chr1:53049-54936    chr1  NC_000001.10            +   
1             chr1:63015-63887    chr1  NC_000001.10            +   
2             chr1:69091-70008    chr1  NC_000001.10            +   
3           chr1:334129-334296    chr1  NC_000001.10            +   
4           chr1:367659-368597    chr1  NC_000001.10            +   
...                        ...     ...           ...          ...   
33292         flmrna->unmapped     ---           NaN          ---   
33293         flmrna->unmapped     ---           NaN          ---   
33294         flmrna->unmapped     ---           NaN          ---   
33295  chr14:96154516-96154596   chr14   NC_000014.8            +   
33296  chr18:14341593-14342513   chr18   NC_000018.9            +   

       RANGE_START  RANGE_STOP  total_probes  \
0            53049       54936             7   
1            63015       63887            31   
2            69091       70008            24   
3           334129      334296             6   
4           367659      368597            36   
...            ...         ...           ...   
33292            0           0            39   
33293            0           0            33   
33294            0           0             6   
33295     96154516    96154596             4   
33296     14341593    14342513             9   

                                         gene_assignment  \
0                                                    ---   
1                                                    ---   
2      NM_001005240 // OR4F17 // olfactory receptor, ...   
3      ENST00000388975 // SEPT14 // septin 14 // 7p11...   
4      NM_001005277 // OR4F16 // olfactory receptor, ...   
...                                                  ...   
33292                                                ---   
33293                                                ---   
33294                                                ---   
33295                                                ---   
33296                                                ---   

                                         mrna_assignment          category  \
0                                                    ---               ---   
1      ENST00000328113 // ENSEMBL // cdna:pseudogene ...              main   
2      NM_001005240 // RefSeq // Homo sapiens olfacto...              main   
3      ENST00000388975 // ENSEMBL // Septin-14 gene:E...              main   
4      NM_001005277 // RefSeq // Homo sapiens olfacto...              main   
...                                                  ...               ...   
33292  NM_002551.3 // --- // Homo sapiens olfactory r...  flmrna->unmapped   
33293  NM_018396.2 // --- // Homo sapiens methyltrans...  flmrna->unmapped   
33294  NM_001040145.1 // --- // Homo sapiens chromoso...  flmrna->unmapped   
33295                                                ---               ---   
33296                                                ---               ---   

      gene_symbol  
0            None  
1            None  
2          OR4F17  
3          SEPT14  
4          OR4F16  

In [22]:
platform_df.to_csv("/content/drive/MyDrive/BIO_MEDICAL_PROJECT/obesity_gene_analysis/data/platform_df.csv")

# Metadata_df 데이터프레임

In [12]:
!pip install GEOparse

In [23]:
import GEOparse

# GEO accession ID가 "GSE77962"인 데이터셋을 다운로드 및 파싱하여 gse 객체 생성
# 다운로드된 파일은 ./ 디렉토리에 저장됨
gse = GEOparse.get_GEO("GSE77962", destdir="./")

# 샘플 메타데이터를 저장할 빈 리스트 생성
metadata = []

# gse.gsms는 각 GSM(샘플)에 대한 정보를 담고 있는 dictionary
for gsm_id, gsm in gse.gsms.items():
    # 각 샘플의 GSM ID를 포함한 딕셔너리 생성
    sample_info = {"gsm_id": gsm_id}

    # 샘플의 특성 정보("characteristics_ch1")를 하나씩 순회
    for line in gsm.metadata.get("characteristics_ch1", []):
        # 특성 정보는 보통 "key: value" 형태로 되어 있음
        if ":" in line:
            # ":" 기준으로 key와 value 분리 (최초의 ':'만 기준)
            key, value = line.split(":", 1)

            # key 전처리: 공백 -> 언더스코어, 소문자로 통일, 괄호 제거 등
            key = key.strip().lower().replace(" ", "_").replace("(", "").replace(")", "")

            # value는 앞뒤 공백 제거
            value = value.strip()

            # 전처리한 key-value 쌍을 sample_info에 추가
            sample_info[key] = value

    metadata.append(sample_info)

metadata_df = pd.DataFrame(metadata)

16-Jun-2025 00:26:25 DEBUG utils - Directory ./ already exists. Skipping.
DEBUG:GEOparse:Directory ./ already exists. Skipping.
16-Jun-2025 00:26:25 INFO GEOparse - Downloading ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE77nnn/GSE77962/soft/GSE77962_family.soft.gz to ./GSE77962_family.soft.gz
INFO:GEOparse:Downloading ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE77nnn/GSE77962/soft/GSE77962_family.soft.gz to ./GSE77962_family.soft.gz
100%|██████████| 47.5M/47.5M [01:32<00:00, 538kB/s] 
16-Jun-2025 00:28:01 DEBUG downloader - Size validation passed
DEBUG:GEOparse:Size validation passed
16-Jun-2025 00:28:01 DEBUG downloader - Moving /tmp/tmpgdh_tuxi to /content/GSE77962_family.soft.gz
DEBUG:GEOparse:Moving /tmp/tmpgdh_tuxi to /content/GSE77962_family.soft.gz
16-Jun-2025 00:28:01 DEBUG downloader - Successfully downloaded ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE77nnn/GSE77962/soft/GSE77962_family.soft.gz
DEBUG:GEOparse:Successfully downloaded ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE77nnn/GSE77

In [24]:
metadata_df

gsm_id subject_id                                tissue  \
0    GSM2062466          1  abdominal subcutaneous white adipose   
1    GSM2062467          1  abdominal subcutaneous white adipose   
2    GSM2062468          1  abdominal subcutaneous white adipose   
3    GSM2062469          2  abdominal subcutaneous white adipose   
4    GSM2062470          2  abdominal subcutaneous white adipose   
..          ...        ...                                   ...   
147  GSM2062613         56  abdominal subcutaneous white adipose   
148  GSM2062614         59  abdominal subcutaneous white adipose   
149  GSM2062615         59  abdominal subcutaneous white adipose   
150  GSM2062616         61  abdominal subcutaneous white adipose   
151  GSM2062617         61  abdominal subcutaneous white adipose   

                 treatment                  time_point     sex age_yrs  \
0    very-low-calorie diet              at study start  female      46   
1    very-low-calorie diet    after weight loss period  female      46   
2    very-low-calorie diet  after weight stable period  female      46   
3    very-low-calorie diet              at study start  female      45   
4    very-low-calorie diet    after weight loss period  female      45   
..                     ...                         ...     ...     ...   
147       low-calorie diet  after weight stable period    male      56   
148       low-calorie diet              at study start    male      67   
149       low-calorie diet    after weight loss period    male      67   
150  very-low-calorie diet              at study start    male      52   
151  very-low-calorie diet  after weight stable period    male      52   

    height_cm weight_kg bmi_kg/m2 body_fat_%  
0         167     83.38      29.9       39.9  
1         167     75.16     26.95       35.8  
2         167     73.79     26.46       32.4  
3         164     80.94     30.09       46.7  
4         164     71.92     26.74         43  
..        ...       ...       ...        ...  
147       171     89.98     30.77       36.5  
148     168.5     82.42     29.03       28.2  
149     168.5     75.01     26.42       21.2  
150       182     94.26     28.46       31.1  
151       182      88.5     26.72       26.6  

[152 rows x 11 columns]

In [25]:
metadata_df.to_csv("/content/drive/MyDrive/BIO_MEDICAL_PROJECT/obesity_gene_analysis/data/metadata_df.csv")

### * 비만 관련 유전자 개수 몇 개?
GWAS Catalog, OMIM, DisGeNET 등에서 **비만(BMI, waist circumference, obesity 등)**과 관련 있다고 보고된 유전자는 수백 개 이상.


### * 10개 정도만 골라도 괜찮은 이유

| 이유                   | 설명                                                                                        |
| -------------------- | ----------------------------------------------------------------------------------------- |
| **1. 대표성 있는 상위 유전자** | FTO, MC4R, LEP 등은 **거의 모든 GWAS에서 등장**하는 핵심 유전자입니다.                                        |
| **2. 예측 모델의 목적**     | 정량적 해석(XAI 기반 SHAP)과 **설명 가능성 확보**를 목표로 할 경우, **적은 수의 강력한 feature**를 쓰는 것이 오히려 해석에 유리합니다. |
| **3. 학습 데이터 크기**     | 현재 subject 수가 53명이라 너무 많은 유전자를 쓰면 오히려 과적합 위험이 있습니다.                                       |

### * non-obesity 유전자 발굴 시 정확성

SHAP은 실제로 **각 유전자 발현량과 감량량/저항성**의 상관성을 통해 영향을 분석
따라서 비만 관련 여부는 SHAP은 모름. 이건 나중에 해석할 때 연구자가 확인하는 것. SHAP 상위 유전자 중 비만유전자 아닌 것 추출	“잠재적 신규 비만 예측 유전자” 후보→ 논문 조사, KEGG pathway 매핑	기능 해석을 통한 생물학적 의미 부여|


### * 비만 유전자 12개 요약 표

| 유전자명       | 기능 설명                                                | 주요 경로                                     | 대사 관련 역할                             | 대표 논문 (PMID)                                               |
| ---------- | ---------------------------------------------------- | ----------------------------------------- | ------------------------------------ | ---------------------------------------------------------- |
| **FTO**    | **m6A RNA demethylase**. 에너지 섭취 조절, 지방세포 분화          | **mTOR, AMPK, GH/IGF-1**                  | 식욕 증가, 칼로리 섭취 증가, 체중 증가 유도           | [PMID: 17632544](https://pubmed.ncbi.nlm.nih.gov/17632544) |
| **MC4R**   | **Melanocortin 4 receptor**. 시상하부에서 식욕 억제 역할         | **Leptin-MSH pathway**                    | 식욕 억제 → 체중 유지. 돌연변이 시 비만 발생          | [PMID: 12538646](https://pubmed.ncbi.nlm.nih.gov/12538646) |
| **LEP**    | **Leptin 단백질** 생성. 지방세포에서 분비되어 식욕 억제 신호 전달           | **JAK/STAT, PI3K-AKT**                    | 포만감 유도, 에너지 항상성. 렙틴 결핍 시 병적 비만       | [PMID: 8552186](https://pubmed.ncbi.nlm.nih.gov/8552186)   |
| **LEPR**   | **Leptin receptor**. 시상하부에서 leptin 신호 수용             | **JAK2/STAT3, IRS**                       | 렙틴 저항성 발생 시 체중 증가 → 비만 유발            | [PMID: 9435158](https://pubmed.ncbi.nlm.nih.gov/9435158)   |
| **ADIPOQ** | **Adiponectin** 단백질 생성. 항염증성, 인슐린 감수성 증가             | **AMPK, PPAR-α**                          | 체지방 감소 촉진, 인슐린 저항성 개선, 대사 증후군 보호 인자  | [PMID: 11240151](https://pubmed.ncbi.nlm.nih.gov/11240151) |
| **PPARG**  | **PPARγ nuclear receptor**. 지방세포 분화 조절               | **Adipogenesis, lipid metabolism**        | 지방축적 조절, 티아졸리디온계 약물의 표적 → 당뇨 및 비만 치료 | [PMID: 11099450](https://pubmed.ncbi.nlm.nih.gov/11099450) |
| **UCP1**   | **Mitochondrial uncoupling protein**. 갈색 지방에서 열 생성   | **Thermogenesis pathway**                 | **열 발생 통한 지방 연소** → 에너지 소비 증가        | [PMID: 8197456](https://pubmed.ncbi.nlm.nih.gov/8197456)   |
| **UCP2**   | UCP1과 유사하지만 **백색지방/췌장 등 다양한 조직에 존재**                 | **ROS 억제, ATP 생산 조절**                     | 에너지 효율 감소, 비만/당뇨 등 대사 질환 연관          | [PMID: 10888871](https://pubmed.ncbi.nlm.nih.gov/10888871) |
| **BDNF**   | **Brain-Derived Neurotrophic Factor**. 뇌에서 식욕 조절에 영향 | **TrkB-NPY, hypothalamus signaling**      | 식욕 억제, BDNF 결핍 시 과식 유도               | [PMID: 10471511](https://pubmed.ncbi.nlm.nih.gov/10471511) |
| **GNPDA2** | **Glucosamine-6-phosphate isomerase**. 신경조직에서 발현     | **Hexosamine biosynthetic pathway**       | GWAS에서 비만 관련 다수 연관 보고됨 (기전은 불명확)     | [PMID: 18953394](https://pubmed.ncbi.nlm.nih.gov/18953394) |
| **TMEM18** | **Transmembrane protein 18**. 시상하부에서 발현, 세포 이동과 관련   | **Neuronal regulation**                   | GWAS에서 체질량 지수(BMI)와 강한 상관관계          | [PMID: 20031578](https://pubmed.ncbi.nlm.nih.gov/20031578) |
| **NEGR1**  | **Neuronal Growth Regulator 1**. 신경 발달 및 식욕 조절       | **Neural plasticity, appetite signaling** | 뇌 기능 변화 통한 식욕 조절. GWAS 상위 유전자        | [PMID: 19079260](https://pubmed.ncbi.nlm.nih.gov/19079260) |
